### PR TITLE
DOC: submit now makes a bit more sense

### LINKED ISSUE #12: job.submit doc string should reflect _submit
...It is misleading, particularly with parameters.

### PR DESCRIPTION
**Pull request by wasade**
Fixes #12

### CODE REVIEW

File: moi/job.py [hunk 1]
```diff
@@ -154,13 +154,38 @@ def _redis_wrap(job_info, func, *args, **kwargs):
...
+    args : tuple or None
+        Any args for ``f``
+    kwargs : dict or None
+        Any kwargs for ``f``
+
+    Returns
+    -------
+    tuple, (str, str, AsyncResult)
+        The job ID and the parent ID
+    """
```
Comment (adamrp) [inline moi/job.py]: Sweet, gotta love documentation! :beers:

File: moi/job.py [hunk 2]
```diff
@@ -154,13 +154,38 @@ def _redis_wrap(job_info, func, *args, **kwargs):
...
     if isinstance(ctx_name, Context):
         ctx = ctx_name
     else:
         ctx = ctxs.get(ctx_name, ctxs[ctx_default])
-    return _submit(ctx, *args, **kwargs)
+    return _submit(ctx, parent_id, name, url, func, *args, **kwargs)
```
Comment (adamrp) [inline moi/job.py]: Will this affect packages (e.g., qiita) that depend on moi in any way?

**Comment by adamrp** [comment]
:+1: curious if the cahnge to `submit` constitutes an API change that packages depending on MOI will need to accommodate.

File: moi/job.py (hunk 2 above)
Comment (wasade) [inline moi/job.py]: No, as `submit` was an adapter for `_submit`

Comment (adamrp) [inline moi/job.py]: Cool, thanks

File: moi/job.py [hunk 3]
```diff
@@ -154,13 +154,38 @@ def _redis_wrap(job_info, func, *args, **kwargs):
...
+    url : str
+        The handler that can take the results (e.g., /beta_diversity/)
+    func : function
+        The function to execute. Any returns from this function will be
+        serialized and deposited into Redis using the uuid for a key. This
+        function should raise if the method fails.
+    args : tuple or None
+        Any args for ``f``
+    kwargs : dict or None
+        Any kwargs for ``f``
```
Comment (adamrp) [inline moi/job.py]: Looks like `f` should be `func` here, and above

Comment (wasade) [inline moi/job.py]: done

File: moi/job.py [hunk 4]
```diff
@@ -154,13 +154,38 @@ def _redis_wrap(job_info, func, *args, **kwargs):
...
+        function should raise if the method fails.
+    args : tuple or None
+        Any args for ``func``
+    kwargs : dict or None
+        Any kwargs for ``func``
+
+    Returns
+    -------
+    tuple, (str, str, AsyncResult)
+        The job ID and the parent ID
```
Comment (josenavas) [inline moi/job.py]: `The job ID, parent ID and the IPython's AsyncResult object of the job`

**Comment by josenavas** [comment]
Just a small documentation comment, otherwise :+1:

**Comment by wasade** [comment]
I _think_ this may have failed due to spin up time. I can recreate it locally, but only if I run nosetests reasonably soon after starting up the cluster. Very annoying, may have to bump sleep to 15s.

**Comment by josenavas** [comment]
I'm really confused with travis now... It looks like the last run was successful, but for some reason it is not updating all the checks in here. Is that correct?

**Comment by wasade** [comment]
Failures are stochastic. One of the element in the build matrix tanked,
apparently 10s to spin up an ipython cluster is no longer sufficient. Weird.

**Comment by ElDeveloper** [comment]
Maybe worth adding a section where the package retries 3 times and waits
10 s in between retry?

**Comment by wasade** [comment]
when you have a hammer, everything looks like a nail :)

much easier to just bump to 30s.

**Comment by ElDeveloper** [comment]
Haha, fair enough.

**Comment by wasade** [comment]
Well... that seemed to work :)

**Comment by josenavas** [comment]
Yay!

**Pull request merged**